moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp

```diff
@@ -115,43 +128,63 @@ void RobotModelLoader::configure(const Options& opt)
-        if (nh.getParam(prefix + "max_position", max_position))
+        std::string param_name;
+        try
         {
-          if (canSpecifyPosition(jmodel, j))
+          param_name = prefix + "max_position";
+          double max_position;
+          if (node_->get_parameter(param_name, max_position))
           {
-            jlim[j].has_position_limits = true;
-            jlim[j].max_position = max_position;
+            if (canSpecifyPosition(jmodel, j))
+            {
+              jlim[j].has_position_limits = true;
+              jlim[j].max_position = max_position;
+            }
           }
-        }
-        double min_position;
-        if (nh.getParam(prefix + "min_position", min_position))
-        {
-          if (canSpecifyPosition(jmodel, j))
+
+          param_name = prefix + "min_position";
+          double min_position;
+          if (node_->get_parameter(param_name, min_position))
           {
-            jlim[j].has_position_limits = true;
-            jlim[j].min_position = min_position;
+            if (canSpecifyPosition(jmodel, j))
+            {
+              jlim[j].has_position_limits = true;
+              jlim[j].min_position = min_position;
+            }
           }
-        }
-        double max_velocity;
-        if (nh.getParam(prefix + "max_velocity", max_velocity))
-        {
-          jlim[j].has_velocity_limits = true;
-          jlim[j].max_velocity = max_velocity;
-        }
-        bool has_vel_limits;
-        if (nh.getParam(prefix + "has_velocity_limits", has_vel_limits))
-          jlim[j].has_velocity_limits = has_vel_limits;
 
-        double max_acc;
-        if (nh.getParam(prefix + "max_acceleration", max_acc))
+          param_name = prefix + "max_velocity";
+          double max_velocity;
+          if (node_->get_parameter(param_name, max_velocity))
+          {
+            jlim[j].has_velocity_limits = true;
+            jlim[j].max_velocity = max_velocity;
+          }
+
+          param_name = prefix + "has_velocity_limits";
+          bool has_vel_limits;
+          if (node_->get_parameter(param_name, has_vel_limits))
+          {
+            jlim[j].has_velocity_limits = has_vel_limits;
+          }
+
+          param_name = prefix + "max_acceleration";
+          double max_acc;
+          if (node_->get_parameter(param_name, max_acc))
+          {
+            jlim[j].has_acceleration_limits = true;
+            jlim[j].max_acceleration = max_acc;
+          }
+
+          param_name = prefix + "has_acceleration_limits";
+          bool has_acc_limits;
+          if (node_->get_parameter(param_name, has_acc_limits))
+            jlim[j].has_acceleration_limits = has_acc_limits;
+        }
+        catch (const rclcpp::ParameterTypeException& e)
         {
-          jlim[j].has_acceleration_limits = true;
-          jlim[j].max_acceleration = max_acc;
+          RCLCPP_WARN(LOGGER, "When getting the parameter %s: %s", param_name.c_str(), e.what());
```

Comment:
I agree it's an `ERROR` since it detects faulty parameter types in the otherwise valid yaml and requires fixup. Also, missing parameters are still supported like before. I'm just concerned that there is currently no way to handle the error (i.e. returning `false`) and terminating the node/session. But this is still better than silently ignoring faulty values as in the original implementation.